### PR TITLE
Add gallery example showing usage of project

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Although we plan to add inline examples for some of the processing function, I think for ``project`` it's easy to show the usage with a very simple example.

Addresses #1666.

**Preview:**

Great circles

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/gallery/lines/great_circles.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+whose center and end points can be defined via the ``center`` and ``endpoint``
+parameters, respectively.
+"""
+
+import pygmt
+
+fig = pygmt.Figure()
+
+# generate points every 10 degrees along a great circle from 10N,50W to 30N,5W
+points1 = pygmt.project(center="-50/10", endpoint="-5/30", generate=10)
```
Comment (seisman) [inline examples/gallery/lines/great_circles.py:15]:
```suggestion
points1 = pygmt.project(center=[-50, 10], endpoint=[-5, 30], generate=10)
```

File: examples/gallery/lines/great_circles.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# generate points every 750 km along a great circle from 10N,50W to 57.5N,90W
+points2 = pygmt.project(center="-50/10", endpoint="-90/57.5", generate=750, unit=True)
```
Comment (seisman) [inline examples/gallery/lines/great_circles.py:17]:
```suggestion
points2 = pygmt.project(center=[-50, 10], endpoint=[-90, 57.5], generate=750, unit=True)
```

File: examples/gallery/lines/great_circles.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# generate points every 350 km along a great circle from 10N,50W to 68N,5W
+points3 = pygmt.project(center="-50/10", endpoint="-5/68", generate=350, unit=True)
```
Comment (seisman) [inline examples/gallery/lines/great_circles.py:19]:
```suggestion
points3 = pygmt.project(center=[-50, 10], endpoint=[-5, 68], generate=350, unit=True)
```

File: examples/gallery/lines/great_circles.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# generate points every 10 degrees along a great circle from 10N,50W to 30N,5W
+points1 = pygmt.project(center=[-50, 10], endpoint=[-5, 30], generate=10)
+# generate points every 750 km along a great circle from 10N,50W to 57.5N,90W
+points2 = pygmt.project(center=[-50, 10], endpoint=[-90, 57.5], generate=750, unit=True)
+# generate points every 350 km along a great circle from 10N,50W to 68N,5W
+points3 = pygmt.project(center=[-50, 10], endpoint=[-5, 68], generate=350, unit=True)
+
+# create a plot with coast and Mercator projection (M)
+fig.basemap(region=[-100, 0, 0, 70], projection="M12c", frame=True)
+fig.coast(shorelines=True)
```
Comment (seisman) [inline examples/gallery/lines/great_circles.py:23]: Setting `area_thresh` to 5000 may make the map cleaner.
```suggestion
fig.coast(shorelines=True, area_thresh=5000)
```

File: examples/gallery/lines/great_circles.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+"""
+Generate points along great circles
+-----------------------------------
+
+The :meth:`pygmt.project` method can generate points along a great circle
+whose center and end points can be defined via the ``center`` and ``endpoint``
+parameters, respectively.
```
Comment (seisman) [inline examples/gallery/lines/great_circles.py:7]: Please explain a little about the output of `project` (e.g., `r` and `s`).

**Review by seisman — APPROVED** [review]

File: examples/gallery/lines/great_circles.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+"""
+Generate points along great circles
+-----------------------------------
+
+The :meth:`pygmt.project` method can generate points along a great circle
+whose center and end points can be defined via the ``center`` and ``endpoint``
+parameters, respectively. Using the ``generate`` parameter allows to generate
+(*r*, *s*, *p*) points every *dist* units of *p* along a profile as
+output. By default all units (*r*, *s* and *p*) are set to degrees while
+``unit=True`` allows to set the unit for *p* to km.
```
Comment (weiji14) [inline examples/gallery/lines/great_circles.py:10]: Using `unit=True` seems very imprecise. Maybe we should use `unit="k"`, and point out the other units available at https://docs.generic-mapping-tools.org/6.3/project#units?

Comment (seisman) [reply examples/gallery/lines/great_circles.py:10]: > Maybe we should use unit="k", and point out the other units available at docs.generic-mapping-tools.org/6.3/project#units?

But `unit` is an alias of `-Q` option, which can't accept unit strings.

Comment (weiji14) [reply examples/gallery/lines/great_circles.py:10]: Ah ok, I misread the `-Q` option. Please ignore this then.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**